gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/TraversalRing.java

```diff
@@ -62,6 +62,10 @@ public void addTraversal(final Traversal.Admin<A, B> traversal) {
         this.traversals.add(traversal);
     }
 
+    public void setTraversal(final int index, final Traversal.Admin<A, B> traversal) {
```

Comment:
Why do a `setTraversal()`? Why not do a `replaceTraversal()`? Then it will be consistent with usage and there is no indexing `i` stuff needed. And this will clean up your code so you don't need the `for()`-loop with `break` and `i++`.